### PR TITLE
Brings the module Dnn 9.2 compatible

### LINKED ISSUE #5: Muliple errors with depricated api's in DNN 9.2
There are multiple errors with the Announcement module using deprecated API's in DNN 9.2
Below is one example. 

Error: Announcements is currently unavailable. DotNetNuke.Services.Exceptions.ModuleLoadException: Method not found: 'System.String DotNetNuke.Entities.Modules.ModuleController.CacheKey(Int32)'. ---> System.MissingMethodException: Method not found: 'System.String DotNetNuke.Entities.Modules.ModuleController.CacheKey(Int32)'. at DotNetNuke.Modules.Announcements.Announcements..ctor() at ASP.desktopmodules_announcements_announcements_ascx..ctor() at __ASP.FastObjectFactory_app_web_bnqtvai1.Create_ASP_desktopmodules_announcements_announcements_ascx() at System.Web.Compilation.BuildResultCompiledType.CreateInstance() at System.Web.UI.TemplateControl.LoadControl(IWebObjectFactory objectFactory, VirtualPath virtualPath, Type t, Object[] parameters) at System.Web.UI.TemplateControl.LoadControl(VirtualPath virtualPath) at DotNetNuke.UI.ControlUtilities.LoadControl[T](TemplateControl containerControl, String ControlSrc) at DotNetNuke.UI.Modules.WebFormsModuleControlFactory.CreateModuleControl(TemplateControl containerControl, ModuleInfo moduleConfiguration) at DotNetNuke.UI.Modules.ModuleControlFactory.LoadModuleControl(TemplateControl containerControl, ModuleInfo moduleConfiguration) at DotNetNuke.UI.Modules.ModuleHost.LoadModuleControl() --- End of inner exception stack trace ---

### PR DESCRIPTION
**Pull request by valadas**
This version will work from Dnn 8.0.0 up to 9.2.0 (maybe future versions too)
I could not manage to build directly, so I updated the build process to use MSBuild.Community.Tasks
I have bumped the version to 07.02.00 because there was already an unpublished 07.01.00
And I updated documentation.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/DNNCommunity/DNN.Announcements/pull/6%23issuecomment-388223233%22%2C%20%22https%3A//github.com/DNNCommunity/DNN.Announcements/pull/6%23issuecomment-388432995%22%2C%20%22https%3A//github.com/DNNCommunity/DNN.Announcements/pull/6%23issuecomment-388433458%22%2C%20%22https%3A//github.com/DNNCommunity/DNN.Announcements/pull/6%23issuecomment-388439197%22%2C%20%22https%3A//github.com/DNNCommunity/DNN.Announcements/pull/6%23issuecomment-388457710%22%2C%20%22https%3A//github.com/DNNCommunity/DNN.Announcements/pull/6%23issuecomment-388457924%22%2C%20%22https%3A//github.com/DNNCommunity/DNN.Announcements/pull/6%23issuecomment-390452690%22%2C%20%22https%3A//github.com/DNNCommunity/DNN.Announcements/pull/6%23issuecomment-390471905%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/DNNCommunity/DNN.Announcements/pull/6%23issuecomment-388223233%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Would%20close%20%235%20%22%2C%20%22created_at%22%3A%20%222018-05-11T00%3A18%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/6371568%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/valadas%22%7D%7D%2C%20%7B%22body%22%3A%20%22Don%27t%20take%20this%20the%20wrong%20way%20and%20I%20may%20be%20incorrect%20in%20my%20assumption%20here%20but%20I%20think%20you%20are%20actually%20breaking%20this%20module.%20%5Cr%5Cn%5Cr%5CnFrom%20what%20I%20seen%20in%20the%20changed%20files%20history%20you%20are%20removing%20the%20reference%20to%20the%20DotNetNuke.Web%20assembly%20and%20adding%20a%20reference%20to%20the%20DotNetNuke.Web.Deprecated%5Cr%5Cn%5Cr%5CnThat%20just%20doesn%27t%20seem%20to%20make%20any%20sense%20since%20DNN%20actually%20physically%20removed%20the%20API%27s%20that%20were%20marked%20as%20deprecated%20over%20the%20last%20years.%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222018-05-11T17%3A35%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/17751004%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/muddeatr%22%7D%7D%2C%20%7B%22body%22%3A%20%22The%20DotNetNuke.Web%20assembly%20is%20actually%20the%20good%20assembly%20and%20the%20DotNetNuke.Web.Deprecated%20is%20bad%20so%20why%20would%20you%20remove%20the%20good%20and%20add%20the%20bad%20%3F%3F%3F%3F%3F%22%2C%20%22created_at%22%3A%20%222018-05-11T17%3A37%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/17751004%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/muddeatr%22%7D%7D%2C%20%7B%22body%22%3A%20%22You%20may%20be%20right%2C%20I%20did%20it%20to%20keep%20the%20telerik%20date%20pickers%20working%2C%20but%20I%20tried%20it%20only%20on%20an%20upgraded%20site%2C%20I%20have%20to%20check%20on%20a%20clean%209.2%20install%20where%20maybe%20that%20dll%20is%20not%20present.%20I%20will%20update...%20%22%2C%20%22created_at%22%3A%20%222018-05-11T17%3A58%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/6371568%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/valadas%22%7D%7D%2C%20%7B%22body%22%3A%20%22Ok%2C%20so%20I%20have%20checked%20and%20DotNetNuke.Web.Deprecated%20is%20still%20being%20installed%20even%20with%20Dnn%209.2.0%20clean%20install.%20According%20to%20the%20release%20notes%2C%20all%20telerik%20wrappers%20where%20moved%20there%20to%20separate%20them%20from%20the%20core%20in%20order%20to%20get%20rid%20of%20them%20in%20the%20future%2C%20but%20as%20of%20now%2C%20it%20is%20still%20being%20shipped%20and%20works.%20The%20new%20DnnDatePicker%20is%20currently%20in%20the%20Internal%20namespace%20of%20DotNetNuke.Web%20which%20means%20it%20may%20still%20have%20breaking%20changes%20in%20future%20releases.%5Cr%5Cn%5Cr%5CnSo%20basically%2C%20this%20pull%20request%20would%20make%20this%20version%20on%20the%20module%20work%20from%208.0.0%20up%20to%209.2.0%20and%20until%20the%20deprecated%20assembly%20is%20actually%20removed.%5Cr%5Cn%5Cr%5CnSo%20I%20suggest%20merging%20this%20and%20I%20will%20add%20another%20issue%20to%20replace%20the%20date%20and%20time%20pickers%20to%20not%20use%20Telerik%20which%20we%20can%20work%20on%20after.%5Cr%5Cn%5Cr%5CnDoes%20that%20sound%20right%3F%22%2C%20%22created_at%22%3A%20%222018-05-11T19%3A06%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/6371568%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/valadas%22%7D%7D%2C%20%7B%22body%22%3A%20%22The%20issue%20I%20created%20is%20issue%20%237%20%22%2C%20%22created_at%22%3A%20%222018-05-11T19%3A07%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/6371568%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/valadas%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40EPTamminga%20do%20you%20think%20we%20should%20merge%20this%20for%20now%20and%20replace%20the%20date%20picker%20later%20%28still%20works%20up%20to%20Dnn%209.2.0%20using%20the%20deprecated.dll%20namespace%29%20I%20can%20replace%20it%20later%20%28created%20another%20issue%20for%20it%29%2C%20but%20I%20am%20trying%20to%20concentrate%20on%20compatibility%20for%20a%20first%20run%20on%20all%20modules.%20Then%20on%20the%20second%20pass%20we%20can%20take%20care%20of%20bugs%20and%20on%20a%20third%20pass%2C%20improvements/new%20features.%20If%20you%20agree%20please%20merge%20that%2C%20if%20not%2C%20let%20me%20know%20and%20I%20can%20take%20care%20of%20adjusting%20this%20pull%20request%20tomorrow.%20Thanks%22%2C%20%22created_at%22%3A%20%222018-05-20T02%3A18%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/6371568%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/valadas%22%7D%7D%2C%20%7B%22body%22%3A%20%22Better%20is%20an%20enemy%20of%20good%20enough.%20If%20it%20works%20and%20enough%20for%20a%20release%2C%20I%20suggest%20to%20create%20the%20release%20and%20make%20issues%20for%20improvements%20in%20an%20update%20later.%20%22%2C%20%22created_at%22%3A%20%222018-05-20T10%3A38%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/5859855%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/EPTamminga%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/DNNCommunity/DNN.Announcements/pull/6#issuecomment-388223233'>General Comment</a></b>
- <a href='https://github.com/valadas'></a> Would close #5
- <a href='https://github.com/muddeatr'></a> Don't take this the wrong way and I may be incorrect in my assumption here but I think you are actually breaking this module.
From what I seen in the changed files history you are removing the reference to the DotNetNuke.Web assembly and adding a reference to the DotNetNuke.Web.Deprecated
That just doesn't seem to make any sense since DNN actually physically removed the API's that were marked as deprecated over the last years.
- <a href='https://github.com/muddeatr'></a> The DotNetNuke.Web assembly is actually the good assembly and the DotNetNuke.Web.Deprecated is bad so why would you remove the good and add the bad ?????
- <a href='https://github.com/valadas'></a> You may be right, I did it to keep the telerik date pickers working, but I tried it only on an upgraded site, I have to check on a clean 9.2 install where maybe that dll is not present. I will update...
- <a href='https://github.com/valadas'></a> Ok, so I have checked and DotNetNuke.Web.Deprecated is still being installed even with Dnn 9.2.0 clean install. According to the release notes, all telerik wrappers where moved there to separate them from the core in order to get rid of them in the future, but as of now, it is still being shipped and works. The new DnnDatePicker is currently in the Internal namespace of DotNetNuke.Web which means it may still have breaking changes in future releases.
So basically, this pull request would make this version on the module work from 8.0.0 up to 9.2.0 and until the deprecated assembly is actually removed.
So I suggest merging this and I will add another issue to replace the date and time pickers to not use Telerik which we can work on after.
Does that sound right?
- <a href='https://github.com/valadas'></a> The issue I created is issue #7
- <a href='https://github.com/valadas'></a> @EPTamminga do you think we should merge this for now and replace the date picker later (still works up to Dnn 9.2.0 using the deprecated.dll namespace) I can replace it later (created another issue for it), but I am trying to concentrate on compatibility for a first run on all modules. Then on the second pass we can take care of bugs and on a third pass, improvements/new features. If you agree please merge that, if not, let me know and I can take care of adjusting this pull request tomorrow. Thanks
- <a href='https://github.com/EPTamminga'></a> Better is an enemy of good enough. If it works and enough for a release, I suggest to create the release and make issues for improvements in an update later.

<a href='https://www.codereviewhub.com/DNNCommunity/DNN.Announcements/pull/6?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/DNNCommunity/DNN.Announcements/pull/6?approve=1'></a>&nbsp;<a href='https://github.com/DNNCommunity/DNN.Announcements/pull/6'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by erikvb** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/DNNCommunity/DNN.Announcements/pull/6'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by valadas** [comment]
Would close #5

**Comment by muddeatr** [comment]
Don't take this the wrong way and I may be incorrect in my assumption here but I think you are actually breaking this module. 

From what I seen in the changed files history you are removing the reference to the DotNetNuke.Web assembly and adding a reference to the DotNetNuke.Web.Deprecated

That just doesn't seem to make any sense since DNN actually physically removed the API's that were marked as deprecated over the last years.

**Comment by muddeatr** [comment]
The DotNetNuke.Web assembly is actually the good assembly and the DotNetNuke.Web.Deprecated is bad so why would you remove the good and add the bad ?????

**Comment by valadas** [comment]
You may be right, I did it to keep the telerik date pickers working, but I tried it only on an upgraded site, I have to check on a clean 9.2 install where maybe that dll is not present. I will update...

**Comment by valadas** [comment]
Ok, so I have checked and DotNetNuke.Web.Deprecated is still being installed even with Dnn 9.2.0 clean install. According to the release notes, all telerik wrappers where moved there to separate them from the core in order to get rid of them in the future, but as of now, it is still being shipped and works. The new DnnDatePicker is currently in the Internal namespace of DotNetNuke.Web which means it may still have breaking changes in future releases.

So basically, this pull request would make this version on the module work from 8.0.0 up to 9.2.0 and until the deprecated assembly is actually removed.

So I suggest merging this and I will add another issue to replace the date and time pickers to not use Telerik which we can work on after.

Does that sound right?

**Comment by valadas** [comment]
The issue I created is issue #7

**Comment by EPTamminga** [comment]
Better is an enemy of good enough. If it works and enough for a release, I suggest to create the release and make issues for improvements in an update later.

**Comment by muddeatr** [comment]
You're replacing a valid current library with a deprecated library to save a control that will be removed in the future, but you have not addressed the issues of the deprecated API calls at all ????? 👎 

This is some messed up logic. I wish I could help but with this kind of thinking, I'm glad I can't and I will have to build my own from here!

**Comment by EPTamminga** [comment]
Please read my comment. Deprecated issues will be covered in the Future. There is nothing wrong with making the module work on 9.2.0 in the first place. Using de deprecated dll is an intermediate step. If you know how to improve the code yourself please provide a PR. 
If you do not want this version, use a previous version until a new version is released that suites your needs.

**Comment by valadas** [comment]
@muddeatr I think there is some confusion here.

There are multiple APIs that where marked as deprecated (this here is an attribute on methods and classes that actually cause a build warning). This kind of deprecation means for module developers: please use the new improved way of doing this because we plan to remove this API. The deprecation attribute usually has a short message guiding the developer to the new API. In this case Dnn **deleted** almost all of the APIs that where marked deprecated before Dnn7 in Dnn9.2. Using these deprecated APIs in a module, makes it break in Dnn 9.2 because it uses APIs that are now **deleted**.

The dotnetnuke.web.dll library contains many common controls that module developers can use. Telerik and Dnn had a partnership where Dnn module developers could use any Telerik controls for free in their modules as long as Dnn provided a wrapper over the Telerik.dll direct implementation. (The idea here I think was to try to shift support to Dnn for that usage scenario). Now, for a couple of reasons, Dnn decided it would be best to not use Telerik controls in the future, but Dnn and multiple existing modules still rely on these controls. So in order to Develop replacements for those Telerik wrappers, Dnn has to build new equivalents, but it makes for a lot of duplicate control names. So, they have moved all telerik wrappers to dotnetnuke.web.deprecated.dll for all Telerik versions and they are implementing replacements for some of them in dotnetnuke.web.dll.

In the case of this module, the only telerik control in use is a calendar dropdown to select a date and time called DnnDatePicker, so now, their is a DnnDatePicker in the deprecated dll that works but will be removed at some time in a future release. There is another DnnDatePicker in an internal namespace. Things in internal namespaces are usually not totally complete or the developers are hesitating on something, but it usually means to module developers: it works but it may change a lot or be replaced shortly.

So in short, the module is fixed for now to work on the latest release. Is it perfect? no, someone could spend a few hours replacing the datepicker with something that is neither deprecated, nor internal.

Don't forget that there are very few people working on these community modules on a volunteer basis and we have a lot of work to do on other modules since Dnn 9.2 removed a lot of APIs used by these. Also many of them have a very old codebase and it may be better to implement improvements after a code refres. So our priority (at least in my case) right now, is to do the bare minimum to make them work and be upgradable to Dnn 9.2. After that, we can start fixing other minor bugs, and merging in improvements and new features.

**Comment by valadas** [comment]
@EPTamminga Can you publish the release of this pull request so people have something downloadable to use it in Dnn 9.2 please.

**Comment by EPTamminga** [comment]
@valadas I did a fork/clone and tried to build locally. Errors about not finding NuGet package. Tried to fix that, not succesful...

`
Error		NuGet Package restore failed for project DotNetNuke.Announcements: Unable to find version '9.1.1.105' of package 'DotNetNuke.Instrumentation'.
  C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\: Package 'DotNetNuke.Instrumentation.9.1.1.105' is not found on source 'C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\'.
  https://api.nuget.org/v3/index.json: Package 'DotNetNuke.Instrumentation.9.1.1.105' is not found on source 'https://api.nuget.org/v3/index.json'.
  https://www.nuget.org/api/v2/curated-feeds/microsoftdotnet/: Package 'DotNetNuke.Instrumentation.9.1.1.105' is not found on source 'https://www.nuget.org/api/v2/curated-feeds/microsoftdotnet/'.
.			0	
`

**Comment by EPTamminga** [comment]
@Valadas: suggestion. Attach your formal correct install pack of the new announcements to this PR, so I can use that zip file as the release version.

**Comment by valadas** [comment]
[DotNetNuke.Announcements.07.02.00.Install.zip](https://github.com/DNNCommunity/DNN.Announcements/files/2078838/DotNetNuke.Announcements.07.02.00.Install.zip)
[DotNetNuke.Announcements.07.02.00.Source.zip](https://github.com/DNNCommunity/DNN.Announcements/files/2078839/DotNetNuke.Announcements.07.02.00.Source.zip)
There we go, I will check on that build issue though on a clean new visual studio install on a VM to make sure building should be easy for everyone. I will create a separate issue from your comment. Thanks.
